rust/datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -162,7 +164,7 @@ macro_rules! compute_op {
 
 macro_rules! binary_string_array_op_scalar {
     ($LEFT:expr, $RIGHT:expr, $OP:ident) => {{
-        let result = match $LEFT.data_type() {
+        let result: Result<Arc<dyn Array>> = match $LEFT.data_type() {
```

Comment:
I could not find a way to do this type erasure directly in `BinaryExpr::evaluate` (L419, which only had scalar operations on `BooleanArray` before), so I'm doing it here and also on `binary_array_op_scalar` (L244).